tests/global_communication/test.cpp

```diff
@@ -28,12 +38,44 @@ int main(int argc, char **argv) {
     // now add our custom printer
     listeners.Append(new mpi_listener("results_global_communication"));
 
-    // record the local return value for tests run on this mpi rank
-    //      0 : success
-    //      1 : failure
-    auto result = RUN_ALL_TESTS();
+    int return_value = 0;
+    try {
+        auto arg = argv+1;
+        while (*arg) {
+            if (auto comm_size = to::parse_opt<unsigned>(arg, 'd', "dryrun")) {
+                if (*comm_size==0) {
+                    throw to::parse_opt_error(*arg, "must be positive integer");
```

Comment:
`arg` is a `char **`; `*arg` is a `char *` that's the currently examined command line argument. `arg` is going to be `argv+n` for some small `n`.

`to::parse_opt<T>(...)` returns an `optional<T>`: if the parse is successful, it holds the value; if it is unsuccessful is holds `nothing`.

The error here is thrown because the value was successfully parsed (`comm_size` is not `nothing`), but the value is not sensible. `*arg` has the C string with the argument that was parsed.